plugins/woocommerce/src/Admin/PluginsInstallLoggers/AsyncPluginsInstallLogger.php

```diff
@@ -14,13 +14,19 @@ class AsyncPluginsInstallLogger implements PluginsInstallLogger {
 	 */
 	private $option_name;
 
+	/*
+	 * $var int $started_time time when the logger was created.
+	 */
+	private $started_time;
+
 	/**
 	 * Constructor.
 	 *
 	 * @param string $option_name option name.
 	 */
 	public function __construct( string $option_name ) {
-		$this->option_name = $option_name;
+		$this->option_name  = $option_name;
+		$this->started_time = time();
```

Comment:
Nit: Implicitly having the timer start here feels a bit off to me. I think an ideal case would be to have a `start` function in the `PluginsInstallLogger` to complement the `complete` function. This then can be called to start the timer. Still, not a blocker. What do you think?

Reply:
That's a good point. I'll make an update

Reply:
Updated in https://github.com/woocommerce/woocommerce/pull/39921/commits/4be982e4f325b6dbc3852d640b52d7c8e81c6e41

Reply:
@ilyasfoo ✋ Could you please take a look at the update?

Reply:
Changes look good, awesome @moon0326!